add appcode flor detail and update root document (#4)
belong-inc/bbti-api-docs: README.md

If you are interested in the API please contact us through our [contact page](https://about.belong.co.jp/contact) or our sales representatives.

# APIs

## Transactional API

> ### Standard API call flow of BBTI is done through two steps:

Reviewer: ```suggestion
### Standard BBTI transaction flow with our API is processed through following two steps:
```

1. Send device information including the condition so that calculate grade and price.
1. Send user detail to go through eKYC process.

![standard_api_call](./diagrams/standard_api_call/standard_api_call.png)

> ### AppCode API call flow of BBTI is done through three steps:

Reviewer: ```suggestion
### AppCode flow is processed through following three steps:
```

> 1. Get model and price list CSV from bbti.

Reviewer: ```suggestion
1. Find model, price, and related info form a CSV provided from Belong in advance.
```

1. Send device information including the condition so that calculate grade and price.
1. Send user detail to go through eKYC process.

> ![appcode_api_call](./diagrams/appcode_api_call/appcode_api_call.png)

Reviewer: ほかは mermaid であるのにここは画像なのは気になります？

Author: 実は他も画像化してリンク貼ってたんですが、GitHub が mermaid 対応される前の名残なので別で消しておきます

### Pricing

#### Standard flow
Device registration in BBTI API handles basic information (model, storage, sim status) as well as device conditions (status of water damage, LCD, camera, and e.t.c.).

> The API calculate a grade of the device based on the condition, then calculate the price of device.

Reviewer 2: これに関しては最新の `PriceProducts` のエンドポイントを記載した方がいいかなと思いました
(`PriceDevices` は廃止予定という認識なので)

${API_SERVER}/bbti/v1/pricing/products

Author: こちらは今も有効な API なので、 Standard flow を今後どうするかも含め別で調整しようかと思います

```text

> ${API_SERVER}/bbti/v1/pricing/devices

Reviewer: `### AppCode API call flow ~~:` の下にありますがこれは AppCode Flow の一部でしたっけ？下にも `#### App code flow` があり文章の構成がなにかおかしく思いましたが、これが AppCode Flow の一部であれば妥当だとは思います(記憶になかったです)

Author: `### Appcode ~` の下に `### Pricing` がいたのでそちらに掛かっている気持ちでしたがそもそも見づらかったので修正しました

#### App code flow
Device registration in the BBTI API handles basic information (model, storage, SIM status) and device state (status of water damage, LCD, camera, etc.) in the form of predefined codes.

The API calculate a grade of the device based on the condition, then calculate the price of device.

```text
${API_SERVER}/bbti/v1/pricing/codes
```

#### User
User registration in BBTI handles user information for the registered device in previous step.

The user registration starts an eKYC process. User needs to complete the eKYC by filling out forms in the provided URL.

```text
${API_SERVER}/bbti/v1/user-kyc
```

#### Event notification
This endpoint is used to notify an event of BBTI such as application completion.

```text
${API_SERVER}/bbti/v1/transactions/{transaction_id}/event
```

#### Delivery
This endpoint is used to register delivery information to BBTI.

```text
${API_SERVER}/bbti/v1/delivery
```

> ## Pricing Only API (Not Transactional)

Reviewer: この一連の API の Not Transactional という見せ方が気になりました。

まず、分け方としては Transactional 云々ではなく以下のように情報の取得に関わる手段(API/CSV) と、Transaction を登録する手段(APIのみ)、価格を確認するという文脈があります。
「価格を確認するとTransactionになる」という頭でこれを書いているように思えますが、これが一般的とは言えません。
文書の構成としては 情報共有方法 -> 価格の確認方法-> Transaction を Belong に登録する方法にしたほうがわかりやすいのではないでしょうか。

1. Belong のモデル情報を得る方法は２通りある
    1.  CSV
    2. API
2. Belong に Transaction を登録する方法は ２通りある
    1. AppCode
    2. 普通の API (使われる予定は現状ない) 
3. Transaction を作らず価格だけ確認する API がある
